File: Geesemon/ClientApp/src/modules/auth/auth.queries.ts
import {gql} from '@apollo/client';
import {Auth} from './auth.types';

export type IsAuthData = { isAuth: Auth }
export type IsAuthVars = {}

export const IS_AUTH_QUERY = gql`
    query IsAuth {
        isAuth {
            user {
                id
                email
                role
                createdAt
                updatedAt
            }
            token
        }
    }
`;

File: Geesemon.GraphQL/Modules/Auth/AuthQueries.cs
using Geesemon.Database.Models;
using Geesemon.Database.Repositories;
using Geesemon.GraphQL.Abstraction;
using Geesemon.GraphQL.Modules.Auth.DTO;
using GraphQL;
using GraphQL.Types;
using Microsoft.AspNetCore.Http;

namespace Geesemon.GraphQL.Modules.Auth
{
    public class AuthQueries : ObjectGraphType, IQueryMarker
    {
        public AuthQueries(UsersRepository usersRepository, IHttpContextAccessor httpContextAccessor, AuthService authService)
        {
            Name = "AuthQueries";

            Field<AuthResponseType>()
                .Name("isAuth")
                .ResolveAsync(async context => 
                {
                    string userEmail = httpContextAccessor.HttpContext.User.Identity.Name;
                    User currentUser = await usersRepository.GetByEmailAsync(userEmail);
                    return new AuthModel()
                    {
                        Token = authService.GenerateAccessToken(currentUser.Id, currentUser.Email, currentUser.Role),
                        User = currentUser,
                    };
                })
                .AuthorizeWith(AuthPolicies.Authenticated);
        }
    }
}
